server/src/main/java/io/crate/role/DropRoleTask.java

```diff
@@ -60,12 +60,11 @@ public ClusterState execute(ClusterState currentState) throws Exception {
     }
 
     @Override
-    protected WriteRoleResponse newResponse(boolean acknowledged) {
-        return new WriteRoleResponse(acknowledged, alreadyExists);
+    protected AcknowledgedResponse newResponse(boolean acknowledged) {
+        return new AcknowledgedResponse(acknowledged && alreadyExists);
     }
 
-    @VisibleForTesting
-    static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
+    private static boolean dropRole(Metadata.Builder mdBuilder, String roleNameToDrop) {
```

Comment:
Why did you change that? The unit tests now are a bit more cumbersome. Do we gain something to test the `execute` instead of the `dropRole` method?

Reply:
yep, it's fine, it doesn't add too much boilerplate to the tests, thx.